torch_frame/nn/models/excelformer.py

```diff
@@ -37,30 +39,59 @@ def feature_mixup(
         y (Tensor): The target.
         num_classes (int): Number of classes.
         beta (float): The concentration parameter of the Beta distribution.
+            (default: :obj:`0.5`)
+        mixup_type (str): The mixup methods. `none` option means no mixup,
+            options `feature` and `hidden` are `FEAT-MIX` (mixup at feature
+            dimensionality) and `HIDDEN-MIX` (mixup at hidden dimensionality)
+            proposed in ExcelFormer paper. (default: :obj:`none`)
+        mi_scores (Tensor): Mutual information scores only used in the
+            mixup weight calculation for `FEAT-MIX`. (default: :obj:`None`)
 
     Returns:
         x_mixedup (Tensor): The mixedup numerical feature.
         y_mixedup (Tensor): Transformed target of size
             :obj:`[batch_size, num_classes]`
     """
     assert num_classes > 0
+    assert mixup_type in ['none', 'feature', 'hidden']
+
+    beta = torch.tensor(beta, dtype=torch.float32, device=x.device)
     beta_distribution = torch.distributions.beta.Beta(beta, beta)
     shuffle_rates = beta_distribution.sample(torch.Size((len(x), 1)))
-    feat_masks = torch.rand(x.shape, device=x.device) < shuffle_rates
     shuffled_idx = torch.randperm(len(x), device=x.device)
+    assert x.ndim == 3, 'FEAT-MIX or HIDDEN-MIX requires encoded features'
+    if mixup_type == 'feature':
+        assert mi_scores is not None
+        mi_scores = mi_scores.to(x.device)
+        # Hard masks (feature dimension)
+        feat_masks = torch.rand(torch.Size((x.shape[0], x.shape[1])),
+                                device=x.device) < shuffle_rates
+        l1_norm_mi = mi_scores / mi_scores.sum()
+        lmbd = torch.sum(
+            l1_norm_mi.unsqueeze(0) * feat_masks, dim=1, keepdim=True)
+        feat_masks = feat_masks.unsqueeze(2)
+    elif mixup_type == 'hidden':
+        # Hard masks (hidden dimension)
+        feat_masks = torch.rand(torch.Size((x.shape[0], x.shape[2])),
+                                device=x.device) < shuffle_rates
+        feat_masks = feat_masks.unsqueeze(1)
+        lmbd = shuffle_rates
+    else:
+        # No mixup
+        feat_masks = torch.ones_like(x, dtype=torch.bool)
+        lmbd = torch.ones_like(shuffle_rates)
     x_mixedup = feat_masks * x + ~feat_masks * x[shuffled_idx]
 
     y_shuffled = y[shuffled_idx]
     if num_classes == 1:
         # Regression task or binary classification
-        shuffle_rates = shuffle_rates.view(-1, )
-        y_mixedup = shuffle_rates * y + (1 - shuffle_rates) * y_shuffled
+        lmbd = lmbd.squeeze(1)
```

Comment:
Can you clarify what `lmbd` stands for? :) Can there be a better name?

Reply:
Thank you! Yes, I'd like to change the mixup weight name to `lam` which is commonly used in mixup implementation, like repos of [mixup-cifar10-Line 119](https://github.com/facebookresearch/mixup-cifar10/blob/main/train.py#L119) and [Manifold-MixUp-Line 106](https://github.com/vikasverma1077/manifold_mixup/blob/master/supervised/models/resnet.py#L106). Necessary comment will be added for the variable.

Reply:
Yes, `lam` is probably better. Thanks :)